grafana/grafana
# Tempo: fix handling of trace-not-found cases

When in Tempo we handle the case of trace-with-this-id-not-found, we use the `.ErrorString` attribute to report the error. This does not work well anymore, the returned `queryResult` structure will not be considered an error, and in the browser we will not get the errorstring that we are sending here. Instead, the `.Error` attribute has to be used.

how to test:

1. in explore-mode enter a non-existing trace-id and run the query. you should get an error-message.
2. go to configuring a tempo data source, press the [TEST] button. it should report the data-source working. (the reason why this happened:  the way the TEST-button works is that we search for a trace with a missing id, and we check the returned error-message if it is the awaited error message. but currently no errors came back, so grafana thought the data-source is not working).
